In [1]:
import numpy as np
import pandas as pd
import statistics
%matplotlib inline

In [42]:
name= ['Greg','Marcia','Peter','Jan','Bobby','Cindy','Cousin Oliver']
brady_bunch = pd.DataFrame(index=name)
brady_bunch['age'] = [14,12,11,10,8,6,8]

### MEAN
print('The average age of the Children was '+str(np.mean(brady_bunch['age'])))

### Median
print('The median age of the Children was '+str(np.median(brady_bunch['age'])))

##Mode
(values, counts) = np.unique(brady_bunch['age'], return_counts=True)
ind = np.argmax(counts)
print('The Mode of the Childen\'s ages was '+str(values[ind]))

##Standard Deviation and Error
print('The Standard Deviation is '+str(np.std(brady_bunch['age']))+' with a Standard Error of '+str(np.std(brady_bunch['age'] ,ddof=1) / np.sqrt(len(brady_bunch['age']))))

### Variance 
print('The varience of the ages is '+str(brady_bunch['age'].var()))

### how do i set the the number to be two decmial and than covnert to a string ?? Do I need to convert to a string?

The average age of the Children was 9.857142857142858
The median age of the Children was 10.0
The Mode of the Childen's ages was 8
The Standard Deviation is 2.531435020952764 with a Standard Error of 1.0334540197243192
The varience of the ages is 7.476190476190475


Question 2: 
I would choose mean of the children. Although the mean and the median are very similar I feel like the mode being below ten
does a better job of showing how young the group of children is. 

I would go with the standard deviation. The variance in age is quite large with one chold being 6 and the othe rbeing 14. I feel
that the standard deviation give a better look at the group as a whole


In [51]:
brady_bunch.at['Cindy', 'age']= 7
### MEAN
print('The average age of the Children was '+str(np.mean(brady_bunch['age'])))

### Median
print('The median age of the Children was '+str(np.median(brady_bunch['age'])))

##Mode
(values, counts) = np.unique(brady_bunch['age'], return_counts=True)
ind = np.argmax(counts)
print('The Mode of the Childen\'s ages was '+str(values[ind]))

##Standard Deviation and Error
print('The Standard Deviation is '+str(np.std(brady_bunch['age']))+' with a Standard Error of '+str(np.std(brady_bunch['age'] ,ddof=1) / np.sqrt(len(brady_bunch['age']))))

### Variance 
print('The varience of the ages is '+str(brady_bunch['age'].var()))

The average age of the Children was 10.0
The median age of the Children was 10.0
The Mode of the Childen's ages was 8
The Standard Deviation is 2.32992949004287 with a Standard Error of 0.9511897312113418
The varience of the ages is 6.333333333333333


Question 3:
    We saw the mode ,standard Deviation and variance change but the median and mode stayed the same. 

In [67]:
name_change= ['Greg','Marcia','Peter','Jan','Bobby','Cindy','Jessica']
brady_bunch2 = pd.DataFrame(index=name_change)
brady_bunch2['age'] = [14,12,11,10,8,6,1]

### MEAN
print('The average age of the Children was '+str(np.mean(brady_bunch2['age'])))

### Median
print('The median age of the Children was '+str(np.median(brady_bunch2['age'])))

##Mode
(values, counts) = np.unique(brady_bunch2['age'], return_counts=True)
ind = np.argmax(counts)
print('The Mode of the Childen\'s ages was '+str(values[ind]))

##Standard Deviation and Error
print('The Standard Deviation is '+str(np.std(brady_bunch2['age']))+' with a Standard Error of '+str(np.std(brady_bunch['age'] ,ddof=1) / np.sqrt(len(brady_bunch['age']))))

### Variance 
print('The varience of the ages is '+str(brady_bunch2['age'].var()))

### adding and removing elements in a DF unclear

The average age of the Children was 8.857142857142858
The median age of the Children was 10.0
The Mode of the Childen's ages was 1
The Standard Deviation is 4.015276949301485 with a Standard Error of 0.9511897312113418
The varience of the ages is 18.80952380952381


The change in age caused a a shift in all stats outside the median. The Mode came back as one but which I know is untrue so we know that we have no mode. 

In [70]:
magazine = ['TV Guide','Entertainment Weekly','Pop Culture','SciPhi']
df = pd.DataFrame(index=magazine)
df['people_score'] = [.20,.23,.17,.05]
df.describe()


people_score
count      4.000000
mean       0.162500
std        0.078899
min        0.050000
25%        0.140000
50%        0.185000
75%        0.207500
max        0.230000

I would say that an average of around 16% of people were fans of the brady bunch. 